File: shared/types/association.ts
/**
 * A cooperative labour association (or federation) that verifies workers
 * and operates the Admin Dashboard.
 *
 * Phase 6A: corrected to match the actual backend `AssociationPublic`
 * response (`backend/app/schemas/association.py`) exactly — this
 * interface previously had a `pinCodesServed: string[]` field that does
 * not exist on the real backend `Association` model (only `id`,
 * `federation_id`, `name` are modeled; pincode coverage is explicitly
 * not modeled at all, per that model's own docstring) and an optional
 * `federationId?`, when the backend's `federation_id` is always present.
 * Nothing in `mobile/` or `admin/` consumed this interface at the time
 * of this correction, so this is a type-only change with no behavior
 * impact.
 */
export interface Association {
  id: string;
  federationId: string;
  name: string;
  createdAt: string;
  updatedAt: string;
}

File: backend/app/schemas/association.py
"""
Typed response models for the Phase 5E-D `GET /federation/me/associations`
route.

Exposes only the fields that exist on the current, locked Phase 5C
`Association` model (`pincode_coverage` is explicitly not modeled at all,
per that model's own docstring, so there is nothing to omit here).
"""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class AssociationPublic(BaseModel):
    """Safe, public view of an `Association`."""

    model_config = ConfigDict(from_attributes=True, populate_by_name=True)

    id: UUID
    federation_id: UUID = Field(alias="federationId")
    name: str
    created_at: datetime = Field(alias="createdAt")
    updated_at: datetime = Field(alias="updatedAt")


class AssociationListResponse(BaseModel):
    """Paginated envelope for `GET /federation/me/associations`."""

    model_config = ConfigDict(populate_by_name=True)

    items: list[AssociationPublic]
    page: int
    page_size: int = Field(alias="pageSize")
    total: int
